app/models/post.rb

```diff
@@ -1,2 +1,19 @@
 class Post < ApplicationRecord
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes
+  belongs_to :user, class_name: 'User'
+  after_create :update_counter
+
+  def upadate_countr
```

Comment:
- [optional] you can use the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private
eg
```ruby
after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```